stdlib/shlex.pyi

```diff
@@ -36,8 +36,7 @@ class shlex(Iterable[str]):
     def push_token(self, tok: str) -> None: ...
     def read_token(self) -> str: ...
     def sourcehook(self, newfile: str) -> tuple[str, TextIO]: ...
-    # TODO argument types
-    def push_source(self, newstream: Any, newfile: Any = ...) -> None: ...
+    def push_source(self, newstream: str | TextIO, newfile: str | None = ...) -> None: ...
```

Comment:
This method can set `self.infile` to `None`. Can you update the annotation of `infile` to reflect that?

I don't like `TextIO` in general, but it's consistent with what the rest of the stub does, and we shouldn't change it in this PR.

Reply:
Done 👍